### DISCUSSION TITLE
Consolidate database config into general Craft config

### DISCUSSION THREAD

**Discussion by michaelrog** [craftcms/cms#11867] [Ideas]
Another tweak that I think could make the Craft config more straightforward to grok, in the long term:

- Deprecate the Database config file and merge those settings into the General config...
- Call the general Craft config file... `config/craft.php`

**Comment by mmikkel** [comment]
Putting aside the fact that the `db.php` file is already sort of deprecated, since its removal from the Craft starter project: Is managing database and "general" config separately really something people struggle with? I feel the separation is pretty natural.

Merging these configs and renaming `general.php` would break every single Craft install out there, with no actual gain for existing sites. Unless there's some real evidence that this change would be helpful to a significant number of people, I'm going to argue against it.

**Reply by brandonkelly** [reply]
Agree, this feels like a lateral move at best. And possibly would make things _more_ confusing, since  there are a lot less DB settings, so they’re easier to grok.
